packages/react-components/react-combobox/src/components/Combobox/useCombobox.tsx

```diff
@@ -226,6 +215,49 @@ export const useCombobox_unstable = (props: ComboboxProps, ref: React.Ref<HTMLIn
 
   state.root.ref = useMergedRefs(state.root.ref, rootRef);
 
+  /* Set input.onKeyDown here, so we can override the default behavior for spacebar */
+  const defaultOnTriggerKeyDown = state.input.onKeyDown;
+  state.input.onKeyDown = useEventCallback((ev: React.KeyboardEvent<HTMLInputElement>) => {
```

Comment:
I just fixed a bug in Tab where I wasn't merging the onClick callback coming in from props with the handler I was setting.  Do you need to merge the callbacks for onKeyDown here? Or does this component not expose the input to callers?

Reply:
The props handler is already merged into `state.input.onKeyDown`, so I'm conditionally calling specifically the props handler vs. the merged handler w/ the selection logic in lines 251-158.

Let me know if that makes sense to you!